### PR TITLE
Sgf parsing

### PR DESCRIPTION
**Pull request by dander**
I finally got all the tests working.

I added a fair amount more feedback to the test runner to help indicate what is wrong when tests fail. It could be useful for other exercises as well.

I also added an optional test runner argument for the number of tests to run.

One thing I'm a bit uncertain about is that the escaping rules are a bit peculiar (`escaped property` test). I'm not sure how much of it is assumption of other languages that things like newlines will be represented like `\n`, but it's a bit awkward for Red. I'm not sure if it would make more sense to convert them to `^/` in the test cases instead...

### CODE REVIEW

File: exercises/practice/sgf-parsing/.meta/config.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+	"blurb": "Parsing a Smart Game Format string.",
+	"authors": [
+		"loziniak",
+		"kickass"
```
Comment (loziniak) [inline exercises/practice/sgf-parsing/.meta/config.json:5]: please, add your GitHub nickname here!

Comment (dander) [reply exercises/practice/sgf-parsing/.meta/config.json]: Thanks! Totally missed that.

File: exercises/practice/sgf-parsing/sgf-parsing.red [hunk 2]
```diff
@@ -3,9 +3,9 @@ Red [
 	author: "" ; you can write your name here, in quotes
 ]
 
-parse: function [
+parse-sgf: function [
 	encoded
 ] [
-	cause-error 'user 'message ["You need to implement this function."]
+	do make error! "You need to implement this function."
```
Comment (loziniak) [inline exercises/practice/sgf-parsing/sgf-parsing.red:9]: Do you think it's generally better method of generating an error? I'm wondering if we should change it in the template and all previous exerises?

Comment (dander) [reply exercises/practice/sgf-parsing/sgf-parsing.red:9]: I don't especially like either version, but having been recently trying to wrap my head around errors myself, I find the `do make!` version easier to understand. First create the error, then activate it. It's still awkward though. I think `cause-error` seems more intentioned as an internal function. For the first two arguments, it only really makes sense to use `'user 'message`, but it's impossible to figure it out just looking at the usage for the function, so they are just extra boilerplate. It just seems more confusing to me.

**Pull request merged**